pennylane/operation.py

```diff
@@ -559,28 +558,67 @@ def hyperparameters(self):
         self._hyperparameters = {}
         return self._hyperparameters
 
-    @staticmethod
-    def decomposition(*params, wires):
-        """Defines a decomposition of this operator into products of other operators.
+    def decomposition(self):
+        """The decomposition of the Operator into a product of more fundamental gates.
 
-        Args:
-            params (tuple[float, int, array]): operator parameters
-            wires (Union(Sequence[int], Wires)): wires the operator acts on
+        .. math:: O = O_1 O_2 \dots O_n
+
+        Returns ``None`` is this Operator does not define its decomposition.
+
+        .. note::
+            By default, this method calls the static method ``compute_decomposition``.
+            Unless the ``compute_decomposition`` has a custom signature, this method should
+            not be overwritten.
 
         Returns:
-            list[Operation]
+            tuple[Operator]: The decomposition of the Operator into lower level operations
+
+        **Example:**
+
+        >>> qml.IsingXX(1.23, wires=(0,1)).decomposition()
+        (CNOT(wires=[0, 1]), RX(1.23, wires=[0]), CNOT(wires=[0, 1]))
+
         """
-        raise NotImplementedError
+        return self.compute_decomposition(*self.parameters, self.wires, **self.hyperparameters)
+
+    @staticmethod
+    def compute_decomposition(*params, wires=None, **hyperparameters):
+        """Determine the Operator's decomposition for specified parameters, wires,
+        and hyperparameters. The decomposition defines an Operator as a product of
+        more fundamental gates:
 
-    def decompose(self):
-        """Decomposes this operator into products of other operators.
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        ``compute_decomposition`` is a static method and can provide the decomposition of a given
+        operator without creating a specific instance. The instance method ``decomposition`` uses
+        this method and the instance's variables.
```

Comment:
```suggestion
        operator without creating a specific instance.
```

I would not even worry to mention it here? No strong conviction though!